tests/base/model_valid_epoch_ends.py

```diff
@@ -43,5 +43,7 @@ def _mean(res, key):
             val_acc_mean /= len(outputs)
 
         metrics_dict = {'val_loss': val_loss_mean.item(), 'val_acc': val_acc_mean.item()}
-        results = {'progress_bar': metrics_dict, 'log': metrics_dict}
-        return results
+        result = metrics_dict.copy()
```

Comment:
why the copy here?

Reply:
Without copying the metric dict, result and metric dict reference the same object. So adding metric dict to result['progress_bar'] would also change metric_dict. If then adding metric dict to result['log'], result['log']['progress_bar'] would exist and cause errors in tests in my machine. 

First I reused the metric_dict by
```
metric_dict['progress_bar'] = metric_dict
metric_dict['log'] = metric_dict
return metric_dict
```
But this is wrong and leads to the same error.